debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/RecoverableConnectionHolder.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.storage.jdbc;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.SQLRecoverableException;
+import java.time.Duration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.storage.jdbc.offset.JdbcOffsetBackingStore;
+import io.debezium.util.DelayStrategy;
+
+public class RecoverableConnectionHolder implements AutoCloseable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JdbcOffsetBackingStore.class);
+    private final String url;
+    private final String user;
+    private final String pwd;
+    private final Duration waitRetryDelay;
+    private final int maxRetryCount;
+
+    private Connection conn;
+
+    public RecoverableConnectionHolder(String url, String user, String pwd, Duration waitRetryDelay, int maxRetryCount) throws SQLException {
+        this.url = url;
+        this.user = user;
+        this.pwd = pwd;
+        this.waitRetryDelay = waitRetryDelay;
+        this.maxRetryCount = maxRetryCount;
+
+        createConnection();
```

Comment:
Should it be called here? It will fail away immediatelly without tying to reconnect, is it expected?

Reply:
Are you sure? IMHO is an exception is thrown it gets propagated from the constructor.

Reply:
oh right, thanks for the catch